.github/workflows/docker-builds.yaml

```
@@ -0,0 +1,82 @@
+name: Build prefect docker images
+
+# Note: Conda support for 3.11 is pending. See https://github.com/ContinuumIO/anaconda-issues/issues/13082
+
+on:
+  # On each commit merged into main, push sha and branch patterns
+  push:
+    branches: main
+  # On workflow_dispatch, push sha and branch patterns
+  workflow_dispatch:
+
+jobs:
+  publish-docker-images:
+    name: Build Docker images
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        flavor:
+          - ""
+          - "-conda"
+        python-version:
+          - "3.7"
+          - "3.8"
+          - "3.9"
+          - "3.10"
+          - "3.11"
+        exclude:
+          # Not yet supported, see note at top
+          - flavor: "-conda"
+            python-version: "3.11"
+
+    steps:
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v2
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
+
+      - name: Login to DockerHub
+        uses: docker/login-action@v2
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Generate image metadata
+        id: metadata
+        uses: docker/metadata-action@v4
+        with:
+          images: prefecthq/prefect-dev
+          # first tag generated here will be used in CACHE_IMAGE below
+          tags: |
+            type=raw,enable=${{ github.event_name == 'release' }},value=${{ github.ref_name }},suffix=-python${{ matrix.python-version }}
```

This will be skipped since this is on push/dispatch for now, right?

---

thanks for pointing this one out, this is not intended. it should be negated, because `ref_name` points to a tag on release event and not a branch. The `git rev-parse` setup from before would return `HEAD` because a release event is a detached branch so is also unusable. it seems to be hard to get out which branch the release was made from.

```suggestion
            type=raw,enable=${{ github.event_name != 'release' }},value=${{ github.ref_name }},suffix=-python${{ matrix.python-version }}
```

this means there will be no cache hit on a release event, because the CACHE_IMAGE would point to the sha pattern instead of the branch pattern, and we dont have `buildcache` for the sha pattern. fix would be to hardcode a buildcache tag in the `cache-from` and `cache-to` based on `github.matrix`. so not getting it from `metadata-action`.

---

I don't think we need to know what branch releases are made from, we can just always pull from the cache on `main`?

---

yes fine by me, will hardcode :+1:

---

[157dd9b](https://github.com/PrefectHQ/prefect/pull/7900/commits/157dd9bfae12ef20ef673f3f9e000d40e070bf14)